### PR TITLE
[ModuleInterface] Guard layout based prespecializations in swiftinter…

### PR DESCRIPTION
**Pull request by drexin**
…face files

rdar://107269447

To allow compilers that don't have this feature enabled to parse interface files that contain declarations that use layout based prespecializations, it has to be guarded.

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

File: test/ModuleInterface/layout_pre_specialization.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-swift-frontend -enable-experimental-feature LayoutPrespecialization -typecheck -module-name LayoutPrespec -emit-module-interface-path %t/LayoutPrespec.swiftinterface %s
```
Comment (tshortli) [inline test/ModuleInterface/layout_pre_specialization.swift:3]: For module interface tests in general I recommend using the `%target-swift-emit-module-interface` and `%target-swift-typecheck-module-from-interface` lit substitutions as they set up a few options that should be used in conjunction with producing an interface like `-enable-library-evolution` and `-swift 5`.

**Review by tshortli — APPROVED** [review]

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test macos

**Comment by drexin** [comment]
@swift-ci smoke test macos

**Pull request merged**